diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index fcc7e1283..ce71d3683 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -262,6 +262,8 @@ if (target_os == "android") {
   _default_toolchain = "//build/toolchain/aix:$target_cpu"
 } else if (target_os == "zos") {
   _default_toolchain = "//build/toolchain/zos:$target_cpu"
+} else if (target_cpu == "wasm") {
+  _default_toolchain = "//build/toolchain/wasm:emscripten"
 } else {
   assert(false, "Unsupported target_os: $target_os")
 }
@@ -377,7 +379,6 @@ if (is_android) {
 
 if (is_clang && !is_nacl) {
   default_compiler_configs += [
-    "//build/config/clang:find_bad_constructs",
     "//build/config/clang:extra_warnings",
   ]
 }
diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index e49fc28c0..b10a9573d 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -287,6 +287,8 @@ config("compiler") {
     configs += [ "//build/config/aix:compiler" ]
   } else if (current_os == "zos") {
     configs += [ "//build/config/zos:compiler" ]
+  } else if (current_os == "wasm") {
+    configs += [ "//build/config/wasm:compiler" ]
   }
 
   configs += [
